state.go

```diff
@@ -83,20 +87,39 @@ func (r *raftState) setCurrentTerm(term uint64) {
 	atomic.StoreUint64(&r.currentTerm, term)
 }
 
-func (r *raftState) getLastLogIndex() uint64 {
-	return atomic.LoadUint64(&r.LastLogIndex)
+func (r *raftState) getLastLog() (index, term uint64) {
+	r.lastLock.Lock()
+	index = r.lastLogIndex
+	term = r.lastLogTerm
+	r.lastLock.Unlock()
+	return
+}
+
+func (r *raftState) getLastLogIndexOnly() uint64 {
```

Comment:
Looking through the history now I realize you had it this way originally :-)

@ongardie were you worried about being able to more easily find places that don't fetch both together or was this just a style thing? It's not a huge deal either way but `idx, _` seems more idiomatic.